backend/common.go

```diff
@@ -178,6 +178,10 @@ type PluginContext struct {
 	// UserAgent is the user agent of the Grafana server that initiated the gRPC request.
 	// Will only be set if request is made from Grafana v10.2.0 or later.
 	UserAgent *useragent.UserAgent
+
+	// FromService is the service in Grafana that initiated the request.
+	// Will only be set if request is made from Grafana v10.4.0 or later.
+	FromService string
```

Comment:
Is the purpose for this, better logging/debugging?  If so maybe `map[string]string`?  as soon as we ask for what "service" next I want to know what dashboard/panel/etc.

That said -- I'm not convinced we want to add this here -- can the context data be added to the grpc request?

Reply:
For load shedding specifically, it is easier for middleware to act on standard headers than needing to parse/understand the messages

Reply:
The purpose is for plugins to be able to adapt their behaviour based on the service that's making requests. 

The reason I'm adding this is so plugins have context if they're supporting user-auth as the query behaviour will need to change in order to support this auth mechanism in the case that a request is made from a backend service where a users credentials aren't in scope e.g. alerting.

I'm not _as_ familiar with gRPC, it probably could be added there instead but it seemed to be easier for plugins to access via the context (this could be a bad assumption on my part).

The only annoying part of this is it still doesn't allow us to distinguish services that use the typical `query` endpoint e.g. dashboards, explore, reporting, or image render requests. We _could_ add these in the request header but then I guess that could be open to spoofing which would be a problem.

Reply:
> The purpose is for plugins to be able to adapt their behaviour based on the service that's making requests.

😬 😬 this is pretty scary if we want the same request to behave differently depending on what service is asking 😬 😬 

If this is auth related.... we really really need to pass that as identity/token/etc.

Reply:
We're going to use the `User` field from the `PluginContext` as the identifier for if a backend service is making the request. If there are any edge-cases then we can revisit this approach 😊